dpgen2/fp/cp2k.py

```diff
@@ -0,0 +1,191 @@
+import os
+from pathlib import (
+    Path,
+)
+from typing import (
+    List,
+    Optional,
+)
+
+import dpdata
+from dargs import (
+    Argument,
+)
+from dflow.python import (
+    OP,
+    OPIO,
+    Artifact,
+    BigParameter,
+    OPIOSign,
+)
+
+try:
+    from fpop.cp2k import (
+        Cp2kInputs,
+        PrepCp2k,
+        RunCp2k,
+    )
+except ModuleNotFoundError:
+    Cp2kInputs = PrepCp2k = RunCp2k = object
+
+from ..constants import (
+    fp_default_out_data_name,
+)
+
+
+class FpOpCp2kInputs(Cp2kInputs):  # type: ignore
+    @staticmethod
+    def args():
+        doc_inp_file = "The path to the user-submitted CP2K input file."
+        return [
+            Argument("inp_file", str, optional=False, doc=doc_inp_file),
+        ]
+
+
+class PrepFpOpCp2k(OP):
+    @classmethod
+    def get_input_sign(cls):
+        return OPIOSign(
+            {
+                "config": BigParameter(dict),
+                "type_map": List[str],
+                "confs": Artifact(List[Path]),
+            }
+        )
+
+    @classmethod
+    def get_output_sign(cls):
+        return OPIOSign(
+            {
+                "task_names": BigParameter(List[str]),
+                "task_paths": Artifact(List[Path]),
+            }
+        )
+
+    @OP.exec_sign_check
+    def execute(
+        self,
+        ip: OPIO,
+    ) -> OPIO:
+        confs = []
+        # remove atom types with 0 atom from type map
+        # for all atom types in the type map
+        for p in ip["confs"]:
+            for f in p.rglob("type.raw"):
```

Comment:
Optional revision: recommended to use `dpdata.MultiSystems` to handle the confs, then you do not need to manually handle the type mappings.  
```python
ms = dpdata.MultiSystems(type_map=type_map) 
ms.from_deepmd_npy(p, labeled=False)
for ii in range(len(ms)): 
    s = ms[ii]                                                                                                                                              
```

Reply:
Hi @wanghan-iapcm,

Thanks for the suggestion. I prefer to keep the current implementation for now.

Reply:
that's no problem.